### PR TITLE
knex: Added modify to QueryInterface

### PR DESCRIPTION
**Pull request by sarahsporck**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<http://knexjs.org/#Builder-modify>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Taggotty** [comment]
Will solve issue #16919

File: types/knex/index.d.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ declare namespace Knex {
         pluck(column: string): QueryBuilder;
 
         insert(data: any, returning?: string | string[]): QueryBuilder;
+        modify(callback: QueryCallback): QueryBuilder;
```
Comment (mastermatt) [inline types/knex/index.d.ts]: According to the linked docs, this should be `modify(callback: QueryCallback, ...args: any): QueryBuilder;`.

**Review by mastermatt — CHANGES_REQUESTED** [review]
Needs the spread for the optional params.

For the tests, I would also recommend copying the example from the docs right into `knex-tests.ts`.

File: types/knex/index.d.ts (hunk 1 above)
Comment (Taggotty) [reply types/knex/index.d.ts]: Thx. Added.

**Review by mastermatt — APPROVED** [review]

**Pull request merged**